cairo-contracts/packages/apps/src/transfer/types.cairo

```diff
@@ -120,11 +120,12 @@ impl PrefixedDenomDisplay of Display<PrefixedDenom> {
 impl PrefixedDenomKeyImpl of ComputeKey<PrefixedDenom> {
     fn key(self: @PrefixedDenom) -> felt252 {
         let mut key_builder = LocalKeyBuilderImpl::init();
-        let mut trace_path_span = self.trace_path.span();
-        while let Option::Some(path) = trace_path_span.pop_front() {
-            key_builder.append_serde(path);
-        };
-        key_builder.append_serde(self.base);
+        // let mut trace_path_span = self.trace_path.span();
+        // while let Option::Some(path) = trace_path_span.pop_front() {
+        //     key_builder.append_serde(path);
+        // };
+        // key_builder.append_serde(self.base);
+        key_builder.append_serde(self);
```

Comment:
I am not sure why `PrefixedDenom` is hashed manually when it implements `Serde` already.

I am ok to revert this and make it compatible at relayer e2e side if this is necessary.

Reply:
I see. But why is this necessary - having the same key for base denom and prefixed denom with empty prefix ?

We are indexing minted tokens. Their prefixes are always non-empty.